src/jdk.incubator.foreign/share/classes/jdk/internal/foreign/MemoryScope.java

```diff
@@ -26,38 +26,73 @@
+ * be closed after all its children have been closed, at which time any associated
+ * cleanup action is executed (the associated memory segment is freed).
+ * Besides thread-confined checked scopes, {@linkplain #createUnchecked(Object, Runnable, Thread)}
+ * method may be used passing {@code null} as the "owner" thread to create an
+ * unchecked scope that doesn't check for thread-confinement or temporal bounds.
  */
 abstract class MemoryScope {
-    public static final MemoryScope GLOBAL = new Root(null, null);
 
     /**
-     * Creates a root MemoryScope with given ref and cleanupAction.
-     * The returned instance may be published unsafely to and used in any thread, but methods that explicitly state that
-     * they may only be called in "owner" thread, must strictly be called in single thread that has been selected to be the
-     * "owner" thread.
+     * Creates a root MemoryScope with given ref, cleanupAction and current
+     * thread as the "owner" thread.
+     * This method may be called in any thread.
+     * The returned instance may be published unsafely to and used in any thread,
+     * but methods that explicitly state that they may only be called in "owner" thread,
+     * must strictly be called in the thread that created the scope
+     * or else IllegalStateException is thrown.
      *
      * @param ref           an optional reference to an instance that needs to be kept reachable
      * @param cleanupAction an optional cleanup action to be executed when returned scope is closed
      * @return a root MemoryScope
      */
     static MemoryScope create(Object ref, Runnable cleanupAction) {
-        return new Root(ref, cleanupAction);
+        return new Root(Thread.currentThread(), ref, cleanupAction);
+    }
+
+    /**
+     * Creates a root MemoryScope with given ref, cleanupAction and "owner" thread.
+     * This method may be called in any thread.
+     * The returned instance may be published unsafely to and used in any thread,
+     * but methods that explicitly state that they may only be called in "owner" thread,
+     * must strictly be called in given owner thread or else IllegalStateException is thrown.
+     * If given owner thread is null, the returned MemoryScope is unchecked, meaning
+     * that all methods may be called in any thread and that {@link #checkValidState()}
+     * does not check for temporal bounds.
+     *
+     * @param ref           an optional reference to an instance that needs to be kept reachable
+     * @param cleanupAction an optional cleanup action to be executed when returned scope is closed
+     * @param owner         the desired owner thread. If {@code owner == null},
+     *                      the returned scope is <em>not</em> thread-confined and not checked.
+     * @return a root MemoryScope
+     */
+    static MemoryScope createUnchecked(Object ref, Runnable cleanupAction, Thread owner) {
```

Comment:
For some (merely stylistic) reason, I'd prefer to see `owner` coming first in the parameter list